test/functional/api/command_spec.lua

```diff
@@ -124,12 +124,27 @@ describe('nvim_create_user_command', function()
       count = 2,
       reg = "",
     }, exec_lua [=[
-      vim.api.nvim_command([[CommandWithLuaCallback hello my\ friend how\ are\ you?]])
+      vim.api.nvim_command([[CommandWithLuaCallback this is    a\ test]])
       return result
     ]=])
 
     eq({
-      args = 'h\tey',
+      args = [[this   includes\ a backslash: \\]],
+      fargs = {"this", "includes a", "backslash:", "\\"},
+      bang = false,
+      line1 = 1,
+      line2 = 1,
+      mods = "",
+      range = 0,
+      count = 2,
+      reg = "",
+    }, exec_lua [=[
+      vim.api.nvim_command([[CommandWithLuaCallback this   includes\ a backslash: \\]])
```

Comment:
Thanks for looking into this gp.

Just a sanity check: from the fargs docs:

> replaces every pair of backslashes (\\) with one backslash

I see the function does ir as the last `\\` is replaces to only one `\`. Right?

Reply:
As far as I can tell this matches what you get when you use

```vim
func Foo(...)
  echom a:000
endfunc

command -nargs=* Foo call Foo(<f-args>)
```

Then run

```vim
:Foo this   includes\ a backslash: \\
```

and you get `['this', 'includes a', 'backslash:', '\']`.

Reply:
Ah it's still not write, because

```vim
:Foo a\b
```

shows `['a\b']`. So the backslash is preserved in that case.

Reply:
Nice! This makes us match fargs now, although rules are quirky. But also if the command wants the full cmdline they can use the unmodified `args`. 👍